Shopify/hydrogen
# calver with changesets

### WHY are these changes introduced?
Whenever we apply a major version of `hydrogen` or `hydrogen-react`, changesets applies semver which results in incorrect package updates. For example, `2023.1` with a major update, results in `2024.0`. Then we have to end up doing a lot of manual work updating packages which is incredibly risky when dealing with publishing packages.

### WHAT is this pull request doing?
This PR patches changesets to allow us to set custom version functions per package. This will allow us to add a function that will do calver instead of semver to specific packages.

### HOW to test your changes?
1. get this branch
2. `npm run clean-all` to clean your node_modules
3. `npm install`
4. `npm run changeset` and create a changeset file for `hydrogen` and/or `hydrogen-react`
5. `npm run changeset versioning`

check that the versions are updated correctly in `package.json`, etc. For example for `2023-04` check that it updates to `2023-07`
